src/main/java/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,373 @@
+package dao;
+
+import dao.interfaces.BookDao;
+import dao.interfaces.mappers.BookMapper;
+import dao.interfaces.mappers.BookRatingMapper;
+import dto.BookDto;
+import models.Book;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+public class BookDaoImpl implements BookDao {
+    private final Connection connection;
+
+    public BookDaoImpl(Connection connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * Method for finding all books
+     * @return all books in database
+     */
+    @Override
+    public List<BookDto> findAll() {
+        String query = "SELECT title, book_description, date_of_publisment, author_name, author_secondname, author_surname, count  FROM book " +
+                "join copy on book.id = copy.book_id " +
+                "join author a on copy.author_id = a.id";
+
+        try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            BookMapper bookMapper = new BookMapper();
+
+            return bookMapper.rowMapper(resultSet);
+        } catch (SQLException e) {
+            return null;
```

Comment:
you are returning null. Not the best approach. Better to re-throw exception at this point.